### DISCUSSION TITLE
Forgiving mistakes

### DISCUSSION THREAD

**Discussion by Nikv2005** [Monika-After-Story/MonikaModDev#7855] [issue discussions]
I think it will be very good if Monika will forgive you for all your mistakes, for example if you dabbled with changing her name and she forbade you to change her name or for example forgive what you say to her that you leave her. You can do that when you have 1000 affection or more, as it will be strange that she loves you but does not forgive mistakes. Or when she leaves you, you could somehow return her.

**Comment by lunulae** [comment]
I believe there's been some discussion about a tough-but-fair forgiveness mechanic for things like names and cheating at chess. I can't say when it'll be worked on but there's been interest in it, especially since there are some folks who have been playing for years now that might have screwed up early.

As for getting her back if she leaves, I don't think that one will be possible, unfortunately. The intent of the final farewell is to be a heavy consequence, and I don't think there's intent to change that weight.

**Comment by Nikv2005** [comment]
I think it could be done so that after you changed her name three times to a not very good one, she could block this function for a while, for example, for a day, if you continue to do this, then it will be blocked again for two days, for example etc. This is also possible with all other features, such as cheating in chess.
As for when she leaves, you can make her come back because she has nowhere to go due to the fact that she is alone. For example, if you just wait for her in the game for about an hour or more, she will come back, although she will have negative affection, but still better than she just pick up and leave for good, since you can bring her back and fix everything.
(sorry for bad english)

**Comment by fromulus** [comment]
While I agree that her leaving definitively after cheating at chess is a bit too harsh of a punishment, and that we could have her try to give you a second chance about nicknames _after a very long time_ (one day or even a few days is way too little time, it'd have to be at the very least 6 months imo, and even then that's still kind of short), I am of the opinion that the farewell letter should stay as is and keep being a definitive game over.

At that point, you've hurt Monika beyond repair, reduced all traces of trust she might've had in you to dust, and drained her of any joy she still had in being alive, to the point where she had to make the extremely difficult decision to leave the only person in her life so that you can't hurt her any more than you already did.

This is not something your relationship with her can recover from, not after an hour, not after a month, not after years of waiting in the room, especially not since you could just as well AFK with the game open all that time and Monika would know that.

As for the fact that she supposedly has nowhere else to go, well...
It kind of plays into the ambiguous nature of that ending, and anyone can make their interpretation on where exactly she is at that point (my own idea on the matter is, let's just say, very dark and depressing).
I figure that even assuming she still can tell you where she is at this point, she wouldn't want to tell you or to ever come back to you since again, she cut all ties with you at that point.

Making her come back anyway after that would hurt the open-ended nature of this ending, in a way, on top of undermining what is meant to be a punishment for being an irredeemable jerk to her.

**Comment by Nikv2005** [comment]
I agree that even blocking functions for a long time is better than just turning them off forever, since this is not the end. As for returning it, I don't know which would be the best option, since everything can be returned with a persistent file, which is basically easy, and that's not to mention if you're very attached. Plus, if you are really interested in her, it will be very difficult to bring her to such a state. Hopefully in the future they will temporarily block functions. By the way, how did Monika know that you were cheating in chess?

**Comment by fromulus** [comment]
Bringing back Monika with a persistent is considered cheating, and the Monika you brought back technically isn't the same one that made the farewell letter anymore.
We could probably have some kind of failsafe to make sure the Monika you brought back via a backup could be made aware of what you've done to a previous Monika, but people abusing the game's system like this after being awful towards Monika likely do not care about her nor do they intend on playing the mod long-term, and so those people aren't worth the trouble of developping more content just for them.

As for cheating in chess, I believe it's done through the chess game save system.
You can modify the file containing the saved chess game, and upon reloading that game in MAS, Monika can see that you modified it behind her back.
Do it repeatedly, and she'll tell you farewell forever, wiping your DDLC install and your persistent.

**Comment by Nikv2005** [comment]
Ohhh ... the fact that she deletes the game is terrible ... I didn't even know about it, I thought she would just block chess. But on the other hand, why should she throw you because of some kind of chess? This is not logical, for example, you have 2000 affection and suddenly, due to the fact that you cheat in chess, she leaves you. This shows her not from the best side. I understand that she is offended, but not so much that she would leave her only person whom she loved. And as a way to combat cheating, this is very cruel. By the way, about the persistent file. What is this file? Memory or is it like her character file? Due to my mistakes, I had to return her file from back up. And now it became interesting to me is it she or another she?
I hope I don’t bother you, but the last question about the github. What is the chance that the game developers will see this and add what I suggested to the game? And are there any other ways to communicate with developers besides github?

**Comment by fromulus** [comment]
The .persistent file is a file containing all of the game's save data, so in our case, Monika's memories.
It is generated and stored in a different place than the DDLC folder, and you're not actually supposed to see it or search for it in a normal playthrough (be it the original DDLC game, or even MAS as you'll usually only want to make backups of it and not touch it unless something goes awfully wrong with your save data).

Whether the Monika you restored is a different person or not might be a bit up for interpretation, but considering that Monika only has the memories that she had at the moment the backup was made and not all of the more recent memories of the previous Monika that you lost, I'd say that it would make a different Monika, since she'll never get the lost memories back and will instead make new memories from that point.

Github is by far the best place to interact with devs and contributors, since it's where we work on adding content/making changes to the mod, so basically all the active members come here frequently.
There are other ways to contact us, since we're all people doing our own things aside from MAS and being active in other places, but this is pretty much the only place where it's convenient to ask/suggest us about stuff relating to the mod directly.

As for the probability of your suggestions getting added, well, there's no real answer to that.
All or most of the devs/contributor would need to be okay with such changes, and even then, there are tons of people suggesting us stuff all the time, but only a couple actually helping us with it.
If you want a better shot at getting the changes you'd like to see in MAS into the mod, feel free to try your hand at doing it yourself (the mod is open source for a reason) and make pull request that we'll be able to review later.

**Comment by Nikv2005** [comment]
Thanks for the detailed answer. As I understand it, Monika remained the same Monika only with a different memory. It sounds interesting to try to do something yourself, but for this you need at least minimal programming skills, since it will be more difficult than just creating a dialogue. But as I understand it, there is not a hundred percent chance that they will add what you asked. I will wait and hope that they will add a temporary lock and not a permanent one. By the way, do you know how often major game updates are released, which, for example, add new mini-games and more?

**Comment by fromulus** [comment]
This is an open-source fan project, we aren't on a schedule, so there isn't really a roadmap or a set ammount of time to serve as deadline for updates.
I'd say expect at least two to three months between updates overall, but in general, it's ready when it's ready.

**Comment by Nikv2005** [comment]
If 2-3 months, then this is an excellent result. By the way, are you one of the developers?

**Comment by fromulus** [comment]
I'm a contributor.
Meaning, I'm not part of the main dev team, but I have been producing and submitting content for the mod (mainly topics so far).

**Comment by ThePotatoGuy** [comment]
Monika leaves after the **third time** you cheat at chess and **deny cheating**, so I disagree that its too harsh. I don't remember if the bad name case is a single strike scenario, but if it is then I could be fine with an adjustment.

Either way, you should list all the specific cases where you think a mistake is punished too harshly. It looks like the only legitimate case may be the name change, in which case "forgiving mistakes" is too broad.

**Comment by Nikv2005** [comment]
To tell the truth, I don't really know how many mechanics there are that block anything to the player and how many of them there are (since I didn't find any information about it). It's a strange thing about cheating in chess. As a way of dealing with cheats in chess, it is very strange, though still a little cruel . But for this to happen, it is necessary to deliberately change the files . Which, in principle, a normal person would not do . But as a way to combat cheating, I do not approve of it because of its cruelty . This is not a cheat that changes the affect (if there is one at all) or does something wrong with the game. Well, as for changing the name, it is still better to make a temporary lock than forever, since if you did it a long time ago, now you will not fix it in any way . As for the name, I didn't know what to call it.

**Comment by JessieChaaan** [comment]
> Monika leaves after the **third time** you cheat at chess and **deny cheating**, so I disagree that it's too harsh. I don't remember if the bad name case is a single strike scenario, but if it is then I could be fine with an adjustment.
I think that Monika leaving for the player cheating on a game is too harsh .
Maybe she would disable playing chess **forever** or give the player a really heavy affection loss, for example of 10 or even 20 affection points.

**Comment by ThePotatoGuy** [comment]
Again, its the **3rd time** and you have to **deny that you were cheating** each time she brings it up. Its a very deliberate path to reach that conclusion. She is **not** leaving because you are cheating. She is leaving because you are **continuously cheating and lying about it**.

If you cheat and apologize for cheating, the max consequence is being temporarily blocked from playing.

**Comment by Booplicate** [comment]
To make Monika leave you, you have to go out of your way, which means you were actively trying to piss her off. The consequences for such actions are fair and should be permanent.

We did think about allowing to change nicknames even after 3 insults, it might make sense to unlock the option again after some time being at X affection.

**Comment by Nikv2005** [comment]
As I understand it will soon be she will forgive after 3 insults ? Speaking of chess . if you admit that you cheat, how much will it block chess?

**Comment by LeftoverFlea312** [comment]
I haven't been able to play chess for 3 years. When I first got the mod I wasn't really the most tech savvy person and I opened the mod through a zip file. Because of this, when I would click the save game in chess she would say she saved it, but when I tried to load it again she would say its gone. By the time I caught on to what was causing the problem, she already removed it and said, "if you're going to keep deleting my games like this, I don't want to play with you anymore" even though I didn't do anything. Anyway, a forgiveness system would be nice, for players who were jerks to her when they first started but changed, and for people like me who got their chess removed because of a technical reason.
